### PR TITLE
Registry Mappings

### PR DESCRIPTION
**Pull request by NoComment1105**

### CODE REVIEW

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		ARG 3 lifecycle
+	METHOD m_kcwggvui registerDefault (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
+		ARG 2 lifecycle
+		ARG 3 bootstrap
+	METHOD m_mckuzarg registerDefaultWithIntrusiveHolders (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
+		ARG 2 lifecycle
+		ARG 3 bootstrap
+	METHOD m_mwsegmux bootStrap ()V
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
	METHOD m_mwsegmux bootstrap ()V
```

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping [hunk 2]
```diff
@@ -19,6 +19,7 @@ CLASS net/minecraft/unmapped/C_wqxmvzdq net/minecraft/util/registry/DynamicRegis
 		ARG 0 key
 	METHOD m_sloxobqt freeze ()Lnet/minecraft/unmapped/C_wqxmvzdq$C_ggmzysrd;
 	METHOD m_vutgtdyk registries ()Ljava/util/stream/Stream;
+	METHOD m_xcyamnji allRegistriesLifecycle ()Lcom/mojang/serialization/Lifecycle;
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]:
```suggestion
	METHOD m_xcyamnji getLifecycleForAllRegistries ()Lcom/mojang/serialization/Lifecycle;
```

File: mappings/net/minecraft/util/registry/HolderLookup.mapping [hunk 3]
```diff
@@ -8,6 +8,7 @@ CLASS net/minecraft/unmapped/C_vtbxyypo net/minecraft/util/registry/HolderLookup
 	CLASS C_drwjtlvl RegistryLookup
 		METHOD m_kqnidhqe enabledIn (Lnet/minecraft/unmapped/C_czxxrbcp;)Lnet/minecraft/unmapped/C_vtbxyypo;
 			ARG 1 featureFlags
+		METHOD m_vhaoljfo registryLifecycle ()Lcom/mojang/serialization/Lifecycle;
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/HolderLookup.mapping]:
```suggestion
		METHOD m_vhaoljfo getRegistryLifecycle ()Lcom/mojang/serialization/Lifecycle;
```

File: mappings/net/minecraft/util/registry/Registry.mapping [hunk 4]
```diff
@@ -3,6 +3,7 @@ CLASS net/minecraft/unmapped/C_tqxyjqsk net/minecraft/util/registry/Registry
 		ARG 1 dynamicOps
 	METHOD m_aeozypen holders ()Ljava/util/stream/Stream;
 	METHOD m_akxosniq getKey ()Lnet/minecraft/unmapped/C_xhhleach;
+	METHOD m_aotfisde getLifeCycle ()Lcom/mojang/serialization/Lifecycle;
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/Registry.mapping]:
```suggestion
	METHOD m_aotfisde getLifecycle ()Lcom/mojang/serialization/Lifecycle;
```

File: mappings/net/minecraft/util/registry/RegistryLoader.mapping [hunk 5]
```diff
@@ -6,7 +6,21 @@ CLASS net/minecraft/unmapped/C_ratuauki net/minecraft/util/registry/RegistryLoad
 	METHOD m_gwtkbndr loadRegistriesIntoManager (Lnet/minecraft/unmapped/C_tmnrpasf;Lnet/minecraft/unmapped/C_wqxmvzdq;Ljava/util/List;)Lnet/minecraft/unmapped/C_wqxmvzdq$C_ggmzysrd;
 		ARG 2 decodingData
 	METHOD m_ikbfvuhz getPath (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/lang/String;
+		ARG 0 identifier
+	METHOD m_iulkejcw createInfoForNewRegistry (Lnet/minecraft/unmapped/C_eexxncvi;)Lnet/minecraft/unmapped/C_piadqmzm$C_sznlrmbg;
+		ARG 0 registry
+	METHOD m_mcjxwvrf createContext (Lnet/minecraft/unmapped/C_wqxmvzdq;Ljava/util/List;)Lnet/minecraft/unmapped/C_piadqmzm$C_raywzonl;
+	METHOD m_nrmjckec loadregistryContents (Lnet/minecraft/unmapped/C_piadqmzm$C_raywzonl;Lnet/minecraft/unmapped/C_tmnrpasf;Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_eexxncvi;Lcom/mojang/serialization/Decoder;Ljava/util/Map;)V
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/RegistryLoader.mapping]:
```suggestion
	METHOD m_nrmjckec loadRegistryContents (Lnet/minecraft/unmapped/C_piadqmzm$C_raywzonl;Lnet/minecraft/unmapped/C_tmnrpasf;Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_eexxncvi;Lcom/mojang/serialization/Decoder;Ljava/util/Map;)V
```

**Review by OroArmor — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping (hunk 2 above)
Comment (NoComment1105) [reply mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: Iirc this is a record

Comment (OroArmor) [reply mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: its a default method on an interface

Comment (NoComment1105) [reply mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: Ah

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	FIELD f_cmyxjngv RECIPE_TYPE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_ctjnkaxt CHUNK_GENERATOR_WORLDGEN Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_dblqyggx STRUCTURE_PLACEMENT_WORLDGEN Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_ecfhfzpk SENSOR_TYPE Lnet/minecraft/unmapped/C_zogerkic;
+	FIELD f_evwusxvi CAT_VARIANT Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_gmdysfqt ENCHANTMENT Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_gruwefbj HEIGHT_PROVIDER_TYPE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_hmjfmfpz STAT_TYPE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_hmxbehsm CUSTOM_STAT Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_idnjesqk identifier Lnet/minecraft/unmapped/C_ncpywfca;
```
Comment (OroArmor) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
	FIELD f_idnjesqk ROOT_REGISTRY_KEY Lnet/minecraft/unmapped/C_ncpywfca;
```

**Review by OroArmor — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	FIELD f_xgsijaxn ACTIVITY Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_xydlmbbh ATTRIBUTE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_yfjtvpdo BLOCK_PROVIDER_TYPE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_ygwpxigt MATERIAL_RULE_WORLDGEN Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_ytznwikb POINT_OF_INTEREST_TYPE Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_zjshyqas DENSITY_FUNCTION_TYPE_WORLDGEN Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_zrzntavz PAINTING_VARIANT Lnet/minecraft/unmapped/C_tqxyjqsk;
+	FIELD f_zydiinmc TRUNK_PLACER_TYPE_WORLDGEN Lnet/minecraft/unmapped/C_tqxyjqsk;
+	METHOD m_bblrifef registerDefaultWithIntrusiveHolders (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		ARG 2 bootstrap
+	METHOD m_fgyhnhhd freeze ()V
+	METHOD m_gfesnbuh registerSimple (Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_tqxyjqsk;
+		ARG 1 bootstrap
+	METHOD m_hhgwdqyg internalRegister (Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_eexxncvi;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;Lcom/mojang/serialization/Lifecycle;)Lnet/minecraft/unmapped/C_eexxncvi;
+		ARG 1 entry
+		ARG 2 bootstrap
+		ARG 3 lifecycle
+	METHOD m_kcwggvui registerDefault (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+		ARG 2 lifecycle
+		ARG 3 bootstrap
+	METHOD m_mckuzarg registerDefaultWithIntrusiveHolders (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/util/registry/BuiltInRegistries.mapping [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+		ARG 2 lifecycle
+		ARG 3 bootstrap
+	METHOD m_mwsegmux bootstrap ()V
+	METHOD m_myqyyllo registerDefault (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lnet/minecraft/unmapped/C_nusqeapl$C_qpkaxgeh;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltInRegistries.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/util/registry/DefaultMappedRegistry.mapping [hunk 11]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/unmapped/C_qnhgcvtq net/minecraft/util/registry/DefaultMappedRegistry
+	FIELD f_mxvnnuoz identifier Lnet/minecraft/unmapped/C_ncpywfca;
+	FIELD f_zosutomx holderReference Lnet/minecraft/unmapped/C_cjzoxshv$C_rjzpeyec;
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Lifecycle;Z)V
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/DefaultMappedRegistry.mapping]:
```suggestion
	FIELD f_mxvnnuoz id Lnet/minecraft/unmapped/C_ncpywfca;
	FIELD f_zosutomx holderReference Lnet/minecraft/unmapped/C_cjzoxshv$C_rjzpeyec;
	METHOD <init> (Ljava/lang/String;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Lifecycle;Z)V
		ARG 1 id
```

File: mappings/net/minecraft/util/registry/Registries.mapping [hunk 12]
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/unmapped/C_msgswxvc net/minecraft/util/registry/Registries
+	FIELD f_trwzeuus LEVEL_STEM Lnet/minecraft/unmapped/C_xhhleach;
+	METHOD m_gtztdwoq levelStemToLevel (Lnet/minecraft/unmapped/C_xhhleach;)Lnet/minecraft/unmapped/C_xhhleach;
+	METHOD m_hnrinasr createRegistryKey (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
+		ARG 0 identifier
+	METHOD m_oinrvkni levelToLevelStem (Lnet/minecraft/unmapped/C_xhhleach;)Lnet/minecraft/unmapped/C_xhhleach;
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/Registries.mapping]: Do we use level stem for these classes?

File: mappings/net/minecraft/util/registry/Registries.mapping [hunk 13]
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/unmapped/C_msgswxvc net/minecraft/util/registry/Registries
+	FIELD f_trwzeuus LEVEL_STEM Lnet/minecraft/unmapped/C_xhhleach;
+	METHOD m_gtztdwoq levelStemToLevel (Lnet/minecraft/unmapped/C_xhhleach;)Lnet/minecraft/unmapped/C_xhhleach;
+	METHOD m_hnrinasr createRegistryKey (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
+		ARG 0 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/Registries.mapping]:
```suggestion
		ARG 0 id
```

File: mappings/net/minecraft/util/registry/RegistryLoader.mapping [hunk 14]
```diff
@@ -6,7 +6,21 @@ CLASS net/minecraft/unmapped/C_ratuauki net/minecraft/util/registry/RegistryLoad
 	METHOD m_gwtkbndr loadRegistriesIntoManager (Lnet/minecraft/unmapped/C_tmnrpasf;Lnet/minecraft/unmapped/C_wqxmvzdq;Ljava/util/List;)Lnet/minecraft/unmapped/C_wqxmvzdq$C_ggmzysrd;
 		ARG 2 decodingData
 	METHOD m_ikbfvuhz getPath (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/lang/String;
+		ARG 0 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/RegistryLoader.mapping]:
```suggestion
		ARG 0 id
```

File: mappings/net/minecraft/util/registry/ResourceFileNamespace.mapping [hunk 15]
```diff
@@ -6,6 +6,13 @@ CLASS net/minecraft/unmapped/C_iiomeaxy net/minecraft/util/registry/ResourceFile
...
 	METHOD m_avnleqiw findAllMatchingResources (Lnet/minecraft/unmapped/C_tmnrpasf;)Ljava/util/Map;
 	METHOD m_emutrpsr unwrapFilePath (Lnet/minecraft/unmapped/C_ncpywfca;)Lnet/minecraft/unmapped/C_ncpywfca;
+		ARG 1 identifier
 	METHOD m_gfkrsuzx findMatchingResources (Lnet/minecraft/unmapped/C_tmnrpasf;)Ljava/util/Map;
+	METHOD m_gnnhrtlr (Lnet/minecraft/unmapped/C_ncpywfca;)Z
+		ARG 1 identifier
+	METHOD m_ixttgzkp (Lnet/minecraft/unmapped/C_ncpywfca;)Z
+		ARG 1 identifier
 	METHOD m_uxvlnvug wrapToFilePath (Lnet/minecraft/unmapped/C_ncpywfca;)Lnet/minecraft/unmapped/C_ncpywfca;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/ResourceFileNamespace.mapping]:
```suggestion
		ARG 1 id
	METHOD m_gfkrsuzx findMatchingResources (Lnet/minecraft/unmapped/C_tmnrpasf;)Ljava/util/Map;
	METHOD m_gnnhrtlr (Lnet/minecraft/unmapped/C_ncpywfca;)Z
		ARG 1 id
	METHOD m_ixttgzkp (Lnet/minecraft/unmapped/C_ncpywfca;)Z
		ARG 1 id
	METHOD m_uxvlnvug wrapToFilePath (Lnet/minecraft/unmapped/C_ncpywfca;)Lnet/minecraft/unmapped/C_ncpywfca;
		ARG 1 id
```

**Review by IotaBread — APPROVED** [review]

**Pull request merged**